pyproject.toml

```diff
@@ -0,0 +1,88 @@
+[tool.poetry]
+name = "spikeinterface"
+version = "0.94.0.dev1"
+license = "MIT"
+
+authors = ["Cole Hurwitz", "Jeremy Magland", "Alessio Paolo Buccino <alessiop.buccino@gmail.com>", "Matthias Hennig", "Samuel Garcia"]
+maintainers = ["Alessio Paolo Buccino <alessiop.buccino@gmail.com>"]
+
+homepage = "https://github.com/SpikeInterface/spikeinterface"
+description = "Python toolkit for analysis, visualization, and comparison of spike sorting output"
+readme = "README.md"
+
+classifiers = [
+    "Programming Language :: Python :: 3",
+    "License :: OSI Approved :: MIT License",
+    "Operating System :: OS Independent",
+]
+
+[tool.poetry.dependencies]
+python = ">=3.7,<3.10"
+
+probeinterface = "^0.2.6"
+
+neo = "^0.10.0"
+joblib = "^1.1.0"
+tqdm = "^4.62.3"
+numpy = "^1.21"
+
+# scipy = {version = "^1.8.0", optional = true}
+scikit-learn = {version = "^1.0.2", optional = true}
+h5py = {version = "^3.6.0", optional = true}
+matplotlib = {version = "^3.5.1", optional = true}
+networkx = {version = "^2.7", python= ">=3.8", optional = true}
+pandas = [
+    {version = "*", python= "~3.7", optional = true},
+    {version = "^1.4.1", python= ">=3.8", optional = true}
+]
+
+datalad = {version = "^0.15.6", optional = true}
+MEArec = {version = "^1.7.2", optional = true}
+pynwb = {version = "^2.0.0", optional = true}
+sonpy = [
+    {version = "^1.8.1", python= "~3.8", optional = true},
+    {version = "^1.9.1", python= "~3.9", optional = true}
+]
+
+# Containerisation frameworks
+spython = {version = "^0.1.18", optional = true}
+docker = {version = "^5.0.3", optional = true}
+
+# For testing only
+pyopencl = {version = "^2022.1", optional = true}
+loky = {version = "^3.0.0", optional = true}
+tridesclous = {version = "^1.6.5", optional = true}
+
+spyking-circus = {version = "^1.1.0", optional = true}
+herdingspikes = {version = "^0.3.99", optional = true}
+
+[tool.poetry.dev-dependencies]
+pytest = "^7.0.1"
+
+[tool.poetry.extras]
+extractors = ["MEArec", "sonpy", "pynwb"]
+
+full = [
+    "h5py", "pandas", "scipy", "scikit-learn", "matplotlib", "networkx"
+]
+
+docker = ["docker"]
+spython = ["spython"]
+
+test = [
+    "docker", "spython",
```

Comment:
I think we decide to not have docker/singularity installed in github actions. Other wise it will run docker tests which we do not want at the moment.

Reply:
Solution:
- Remove docker/singularity from the test extra
- Docker testing: `pip install spikeinterface[docker]` -> inside container -> `pip install spikeinterface[full]`
- Singularity testing: `pip install spikeinterface[spython]` -> inside container -> `pip install spikeinterface[full]`